### ISSUE TITLE
AWS S3 connection leaking

### ISSUE THREAD

**Issue by jweisman** [medusa-project/cantaloupe#101]
Hi there. Working to implement your very nice image server. Running into the following:

`Timeout waiting for connection from pool`

According to this [issue](https://github.com/aws/aws-sdk-java/issues/269), it can happen if references to S3Objects are not closed. 

Have you run into this? I couldn't see where the S3Object returned by theS3Resolver is closed, but maybe I'm missing something?

**Comment by adolski** [comment]
Hi, that is probably an RTFM failure on my part. Thanks for looking into it. I will try to get a fix committed soon.

**Comment by adolski** [comment]
I did find one place in AmazonS3Resolver where an S3 resource was not being closed. It looks to be fixed based on looking at the code, but there isn't a test for this, so I'm not sure.

The fix was made on the `release/3.3` branch and then merged into `develop`. You are welcome to compile your own build based on one of these (`mvn package -Dmaven.test.skip=true`) and I would be interested to find out if it does indeed work. It would be nice to get this fix into 3.3.1.

**Comment by jweisman** [comment]
I tested it in the `release/3.3 branch`. Looks better now. I did a bunch of simultaneous requests and it always went back to an empty connection pool. 

`0:29:50.920 [qtp1104106489-11] DEBUG o.a.h.i.c.PoolingClientConnectionManager - Connection released: [id: 8][route: {s}->https://almad-test.s3.amazonaws.com][total kept alive: 1; route allocated: 1 of 50; total allocated: 1 of 50]
`

Before this change, the number of allocated connections kept creeping up.

Thanks!

-Josh
